src/main/java/com/libraryproject/librarysystem/controller/BookController.java

```diff
@@ -0,0 +1,32 @@
+package com.libraryproject.librarysystem.controller;
+
+import com.libraryproject.librarysystem.dto.BookResponseDto;
+import com.libraryproject.librarysystem.service.BookService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/books")
+public class BookController {
+    private final BookService bookService;
+
+    @GetMapping("/page")
+    public ResponseEntity<Page<BookResponseDto>> getAllBooks(@RequestParam(required = false) Integer page){
+        int pageNumber = (page != null && page >= 0) ? page : 0;
```

Comment:
imho paginacja jest super pod tym względem że zwracane sa informacje nt stron, np. ich ilość. Dzięki temu uważam nie ma potrzeby zabezpieczać kodu np. przed negatywną lub za dużą stroną, strona zostanie zwrócona pusta, a jeśli konsumer zastanawia się dlaczego to dotanie też informacje że `"totalPages": 1`, a osągając ostatnią stronę ma również flagę `"last": true`